keanu-project/src/main/java/io/improbable/keanu/util/DescriptionUtils.java

```diff
@@ -0,0 +1,61 @@
+package io.improbable.keanu.util;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.bool.BooleanVertex;
+import io.improbable.keanu.vertices.bool.nonprobabilistic.operators.binary.BooleanBinaryOpVertex;
+import io.improbable.keanu.vertices.intgr.probabilistic.BinomialVertex;
+
+import static io.improbable.keanu.util.DescriptionCreator.createDescriptionAllowingLabels;
+
+class DescriptionUtils {
+
+    static String createIfStringDescription(BooleanVertex predicate, Vertex thn, Vertex els, boolean includeBrackets) {
+        StringBuilder builder = new StringBuilder();
+
+        if (includeBrackets) {
+            builder.append("(");
+        }
+        builder.append(createDescriptionAllowingLabels(predicate));
+        builder.append(" ? ");
+        builder.append(createDescriptionAllowingLabels(thn));
+        builder.append(" : ");
+        builder.append(createDescriptionAllowingLabels(els));
+        if (includeBrackets) {
+            builder.append(")");
+        }
+
+        return builder.toString();
+    }
+
+    static String createBinomialDescription(BinomialVertex binomialVertex, boolean includeBrackets) {
```

Comment:
We could use the save/load annotations for random variables and auto generate reasonable looking things for all of them)?